src/merkle/smt/mod.rs

```diff
@@ -453,6 +453,7 @@ pub(crate) trait SparseMerkleTree<const DEPTH: u8> {
 
     /// Constructs a single leaf from an arbitrary amount of key-value pairs.
     /// Those pairs must all have the same leaf index.
+    #[cfg_attr(not(feature = "concurrent"), expect(dead_code))]
     fn pairs_to_leaf(pairs: Vec<(Self::Key, Self::Value)>) -> Self::Leaf;
```

Comment:
Seems like this function is used only when `concurrent` feature is enabled, right? If so, does it actually need to be defined here? Should we move it to the `src/merkle/smt/full/concurrent.rs` module?